src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -452,3 +471,33 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
         )
         df._reader = self
         return df
+
+    def _upload_local_file_to_stage(self, path: str, format: str) -> str:
+        ostype = os.name
+        temp_stage = self._session.get_session_stage()
+        if os.path.exists(path) and os.path.isfile(path):
```

Comment:
Should we also check if the file is the specified format?

Reply:
I think its ok to not check the format? The format should have been checked in other functions